### ISSUE TITLE
Open command always in CLI mode, don't prompt for cli/gui

### ISSUE THREAD

**Issue by freijon** [jarun/nnn#1350]
I use `nnn` via ssh on a VM without GUI. When I open a file with the "o" key, `nnn` always prompts me to choose if I want to open in _cli_ or _gui_ mode. I don't want to see this option at all, I want `nnn` to always assume _cli_ was chosen.

I read the manual and this is what I tried:

- Run `nnn -c`
- Setting ENV `NNN_OPENER=/home/<user>/.config/nnn/plugins/nuke` (GUI = 0)

Am I missing something here?

**Comment by N-R-K** [comment]
You should open files with `enter` or `l`.
`o` is just a fallback when you don't want to use your default opener.

**Comment by freijon** [comment]
And this is also exactly what I do to edit files.
However, I use `o` to run commands like:

- git log
- git add
- git difftool
- rclone
- custom commands from my .local/bin
- ...

running `o` is one of my most common actions... And I always have to select _cli_ first, which becomes annoying.

**Comment by N-R-K** [comment]
You can easily write a plugin for this. https://github.com/jarun/nnn/tree/master/plugins has detailed instructions on how to do it.

**Comment by freijon** [comment]
Indeed, and I do use plugins. But I need to assign plugins to a letter and I need to remember exactly which letter does what. A lot of letters are already used for other stuff. Also, sometimes I like to pass custom arguments/options like git tags to the open-command that I can't hard-code in a plugin. It seems like a workaround for something that could be (seemingly) easily fixed without much effort and side-effects. I thought about an ENV variable `NNN_OPEN_WITH` which can be set to _CLI_ would that be totally out of the question?

**Comment by N-R-K** [comment]
Plugins don't have to be hardcoded, you can read arbitrary command from the cli into a variable/array and execute it. I can't help you any further at this time as I'm a bit busy with other stuff.

**Comment by 0xACE** [comment]
Not near a machine atm, but https://github.com/jarun/nnn/blob/fd998b3045fbc8ef3a97d2f55a80781725f76919/src/nnn.c#L6031-L6042

just change that into

```
mkpath(path, name, newpath);
spawn(tmp, newpath, NULL, NULL, F_CLI);
```

in your local build, you will be able to rebase this on master without much issue whenever you upgrade, otherwise you could add a hook for `getenv(DISPLAY)` or something along those lines to deduce if `GUI` is even a option, but I would read what those flags mean before going so far…

**Comment by freijon** [comment]
You were faster...
I created this patch to resolve it:
```
--- nnn/src/nnn.c	2022-04-29 07:53:32.330059885 +0200
+++ nnn/src/nnn.c	2022-04-29 08:09:37.406099272 +0200
@@ -5928,11 +5928,16 @@
 
 static void handle_openwith(const char *path, const char *name, char *newpath, char *tmp)
 {
-	/* Confirm if app is CLI or GUI */
-	int r = get_input(messages[MSG_CLI_MODE]);
+	char* cli = getenv("NNN_OPEN_WITH_CLI_ONLY");
+	int r = -1;
+	if (cli) r = F_CLI;
+	else {
+		/* Confirm if app is CLI or GUI */
+		r = get_input(messages[MSG_CLI_MODE]);
 
-	r = (r == 'c' ? F_CLI :
-	     (r == 'g' ? F_NOWAIT | F_NOTRACE | F_MULTI : 0));
+		r = (r == 'c' ? F_CLI :
+		     (r == 'g' ? F_NOWAIT | F_NOTRACE | F_MULTI : 0));
+	}
 	if (r) {
 		mkpath(path, name, newpath);
 		spawn(tmp, newpath, NULL, NULL, r);
```
